torch/csrc/jit/codegen/cuda/scheduler/registry.cpp

```diff
@@ -866,8 +866,22 @@ class NoOpScheduler : public SchedulerEntry {
   //! Check if the no-op heuristics apply in given fusion
   static bool canScheduleCompileTime(Fusion* fusion) {
     // Check there're no non-trivial reduction ops.
-    if (!ir_utils::getReductionOps(fusion, true /* ignore_trivial */).empty()) {
-      return false;
+    for (auto reduction :
+         ir_utils::getReductionOps(fusion, true /* ignore_trivial */)) {
+      for (auto input :
+           ir_utils::filterByType<TensorView>(reduction->inputs())) {
+        auto root_dom = input->getRootDomain();
+        auto all_nonzero =
```

Comment:
Should we add the reverse check to the other schedulers?

Reply:
I don't think we need to do so. In this list:
https://github.com/csarofeen/pytorch/blob/40bd463ff010b384c2248e47079f96d80c9dcc9b/torch/csrc/jit/codegen/cuda/scheduler/registry.cpp#L1474-L1483
The NoOp scheduler is the top, so as long as it says yes, other schedulers will not have a chance.